src/lib/Underground.js

```diff
@@ -438,4 +470,63 @@ class AutomationUnderground
 
         return itemsState;
     }
+
+    /**
+     * @brief Check for daily deals opportunities to make more diamonds and trade those that are beneficial
+     */
+    static __internal__tradeDiamonds()
+    {
+        const deals = DailyDeal.list()
+        let dealsDone = 0;
+        let totalProfit = 0;
+        for (const [ i, deal ] of deals.entries())
+        {
+            let fromValue = deal.amount1 * deal.item1.value;
+            let toValue = deal.amount2 * deal.item2.value;
+            // Fake value of source item if it is not a diamond item
+            if (deal.item1.valueType != UndergroundItemValueType.Diamond)
+            {
+                if (Automation.Utils.LocalStorage.getValue(this.Settings.TradeAnyToDiamonds) === "true")
+                {
+                    fromValue = 0;
+                }
+                else
+                {
+                    fromValue = Number.MAX_SAFE_INTEGER;
+                }
+            }
+            // Do not trade if the destination is not diamonds
+            if (deal.item2.valueType != UndergroundItemValueType.Diamond)
+            {
+                toValue = 0;
+            }
+            // Do not trade if the source is locked
+            if(player.mineInventory().find(item=>item.id == deal.item1.id).sellLocked())
+            {
+                fromValue = Number.MAX_SAFE_INTEGER;
```

Comment:
Same thing in this case, you should continue

Reply:
Already implemented in the latest commit